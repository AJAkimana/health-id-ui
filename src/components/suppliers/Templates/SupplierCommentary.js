import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import {
  Typography,
  Grid
} from '@material-ui/core';

const SupplierCommentary = (props) => {
  const { classes, renderTextField, supplier } = props;

  return (
    <Fragment>
      <div className={classes.dividerDiv}>
        <Typography
          paragraph
          variant="h6"
          align="left"
          gutterBottom
          className={classes.dividerHeaders}
        >
          Commentary
        </Typography>
      </div>
      <Grid container spacing={24} className={classes.containerGrid}>
        <Grid item xs={12}>
          {renderTextField(
            classes.descriptionFields, 'commentary', '', supplier.commentary
          )}
        </Grid>
      </Grid>
    </Fragment>
  );
};

SupplierCommentary.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  renderTextField: PropTypes.func.isRequired,
  supplier: PropTypes.shape({
    commentary: PropTypes.string
  }).isRequired
};

export default SupplierCommentary;
