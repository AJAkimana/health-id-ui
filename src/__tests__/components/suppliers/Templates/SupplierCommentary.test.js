import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SupplierCommentary from '../../../../components/suppliers/Templates/SupplierCommentary';

const props = {
  supplier: {
    isApproved: false,
    id: 'AlphaX',
    address: '​​​​​NNPC Towers, Central Business District, Abuja.',
    name: 'Damian Inc',
    mobile: '08137519698',
    rating: '5',
    image:
      'https://justcreative.com/wp-content/uploads/2019/09/brand-strategy-workbook-1.jpg',
    email: 'test@gmail.com',
    tier: 'Manufacturer',
    paymentTerms: 'Daily',
    creditDays: '3 days',
  },
  renderTextField: jest.fn(),
  variables: {},
  classes: {},
  session: {
    me: {
      role: {
        name: 'Master Admin'
      },
      activeOutlet: {
        outletpreference: {
          outletTimezone: {
            name: "Africa/Nairobi"
          }
        }
      }
    }
  }
};

describe('SupplierCommentary', () => {
  it('should render the SupplierCommentary component properly', () => {
    const setupSupplierCommentary = () => {
      const wrapper = shallow(<SupplierCommentary {...props} />)
      return wrapper
    }

    let wrapper;
    wrapper = setupSupplierCommentary();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
