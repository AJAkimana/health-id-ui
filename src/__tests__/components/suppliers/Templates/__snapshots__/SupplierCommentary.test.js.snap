// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierCommentary should render the SupplierCommentary component properly 1`] = `
<Fragment>
  <div>
    <WithStyles(Typography)
      align="left"
      gutterBottom={true}
      paragraph={true}
      variant="h6"
    >
      Commentary
    </WithStyles(Typography)>
  </div>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    />
  </WithStyles(Grid)>
</Fragment>
`;
